pkg/exthttp/transport.go

```diff
@@ -25,3 +45,45 @@ func NewTransport() *http.Transport {
 		ExpectContinueTimeout: 1 * time.Second,
 	}
 }
+
+// DefaultTransport - this default transport is based on the Minio
+// DefaultTransport up until the following commit:
+// https://github.com/minio/minio-go/commit/008c7aa71fc17e11bf980c209a4f8c4d687fc884
+// The values have since diverged.
+func DefaultTransport(config HTTPConfig) (*http.Transport, error) {
+	tlsConfig, err := objstore.NewTLSConfig(&config.TLSConfig)
+	if err != nil {
+		return nil, err
+	}
+
+	if config.InsecureSkipVerify {
+		tlsConfig.InsecureSkipVerify = true
+	}
```

Comment:
@SrushtiSapkale I think this would be the other way around i.e, `tlsConfig.InsecureSkipVerify = config.InsecureSkipVerify` which is why the test might be failing [here](https://github.com/thanos-io/thanos/pull/5447#issuecomment-1169940766). Can you try this out and check? :)